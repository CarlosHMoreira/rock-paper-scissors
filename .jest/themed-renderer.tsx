import React from 'react';
import { create } from 'react-test-renderer';

import Theme from '../src/theme';

export const renderWithTheme = (component: React.ReactNode) => create(<Theme>{component}</Theme>);