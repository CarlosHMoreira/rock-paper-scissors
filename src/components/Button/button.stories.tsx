import React from 'react';
import Button from './index';

export default {
  title: "Button",
}

export const withText = () => (<Button>Play Again</Button>);

export const withBig = () => (<Button big>Play Again</Button>);

export const withOutline = () => (<Button outline>Rules</Button>);

export const withBigAndOutline = () => (<Button big outline>Rules</Button>);
