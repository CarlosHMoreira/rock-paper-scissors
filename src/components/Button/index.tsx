import React from 'react';
import styled from 'styled-components';

export type Props  = {
  outline?: boolean,
  big?: boolean,
  children: React.ReactNode
}

const Button = styled.button<Props>`
  min-width: ${({ big }) => big ? '200px' : '120px' };
  padding: .7rem .7rem .5rem;
  border: 1.5px solid #fff;
  border-radius: 8px;
  background-color: ${({ outline }) => outline ? 'transparent' : '#fff' };
  font-family: BarlowSemiCondensed;
  text-transform: uppercase;
  color: ${({ outline, theme }) => outline ? '#fff' : theme.colors.neutral.darkText };

  &:hover {
    cursor: pointer;
  }
`;

export default Button;