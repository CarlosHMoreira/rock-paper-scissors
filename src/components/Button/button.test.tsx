import React from 'react';
import 'jest-styled-components'

import { renderWithTheme } from '../../../.jest/themed-renderer';

import Button from '.';

describe('Button', () => {
  test('it should match the default button snapshot', () => {
    const component = (<Button>Some content</Button>);
    const snapshot = renderWithTheme(component);
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  test('it should match the big button snapshot', () => {
    const component = (<Button big >Some content</Button>);
    const snapshot = renderWithTheme(component);
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  test('it should match the outline button snapshot', () => {
    const component = (<Button outline >Some content</Button>);
    const snapshot = renderWithTheme(component);
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  test('it should match the big and outline button snapshot', () => {
    const component = (<Button big outline >Some content</Button>);
    const snapshot = renderWithTheme(component);
    expect(snapshot.toJSON()).toMatchSnapshot();
  });
});