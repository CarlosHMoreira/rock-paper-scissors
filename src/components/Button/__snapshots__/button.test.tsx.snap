// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button it should match the big and outline button snapshot 1`] = `
<button
  className=""
>
  Some content
</button>
`;

exports[`Button it should match the big button snapshot 1`] = `
<button
  className=""
>
  Some content
</button>
`;

exports[`Button it should match the default button snapshot 1`] = `
<button
  className=""
>
  Some content
</button>
`;

exports[`Button it should match the outline button snapshot 1`] = `
<button
  className=""
>
  Some content
</button>
`;
